provision/kubernetes/helpers.go

```diff
@@ -1357,3 +1358,43 @@ func ignoreBaseDep(versionedGroup map[int][]deploymentInfo) {
 		}
 	}
 }
+
+func topologySpreadConstraints(labels map[string]string, topologySpreadConstraintRule string) ([]apiv1.TopologySpreadConstraint, error) {
+	if topologySpreadConstraintRule == "" {
+		return nil, nil
+	}
+	var topologySpreadConstraintList []map[string]interface{}
+	err := json.Unmarshal([]byte(topologySpreadConstraintRule), &topologySpreadConstraintList)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSON object for topologySpreadConstraint: %v", err)
+	}
+	var topologySpreadConstraints []apiv1.TopologySpreadConstraint
+	for _, item := range topologySpreadConstraintList {
+		maxSkew, maxSkewExists := item["maxskew"].(float64)
+		topologyKey, topologyKeyExists := item["topologykey"].(string)
+		if !maxSkewExists || !topologyKeyExists {
+			return nil, errors.New("maxskew and topologykey are required in each topologySpreadConstraint")
+		}
+		constraint := apiv1.TopologySpreadConstraint{
+			MaxSkew:     (int32)(maxSkew),
+			TopologyKey: topologyKey,
+			LabelSelector: &metav1.LabelSelector{
+				MatchLabels: filterAppLabels(labels),
+			},
+			WhenUnsatisfiable: apiv1.ScheduleAnyway,
+		}
+		topologySpreadConstraints = append(topologySpreadConstraints, constraint)
+	}
```

Comment:
I think doesn't make sense forcing some default values here - such as WhenUnsatisfiable to "ScheduleAnyway". I'd keep only the inclusion of app's labels in the label selector instead.

```suggestion
	for _, c := range topologySpreadConstraintList {
		if c.LabelSelector == nil {
			c.LabelSelector = &metav1.LabelSelector{}
		}

		if c.LabelSelector.MatchLabels == nil {
			c.LabelSelector.MatchLabels = make(map[string]string)
		}

		for k, v := range filterAppLabels(labels) {
			c.LabelSelector.MatchLabels[k] = v
		}
		
		topologySpreadConstraints = append(topologySpreadConstraints, c)
	}
```